rasa/shared/core/flows/flows_yaml_schema.json

```diff
@@ -0,0 +1,293 @@
+{
+  "type": "object",
+  "required": [
+    "flows"
+  ],
+  "properties": {
+    "version": {
+      "type": "string"
+    },
+    "flows": {
+      "type": "object",
+      "patternProperties": {
+        "^[A-Za-z_][A-Za-z0-9_]*$": {
+          "$ref": "#$defs/flow"
+        }
+      }
+    }
+  },
+  "$defs": {
+    "steps": {
+      "type": "array",
+      "minContains": 1,
+      "items": {
+        "type": "object",
+        "oneOf": [
+          {
+            "required": [
+              "action"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "action": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "collect"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "description":{
+                "type": "string"
+              },
+              "collect": {
+                "type": "string"
+              },
+              "ask_before_filling": {
+                "type": "boolean"
+              },
+              "reset_after_flow_ends": {
+                "type": "boolean"
+              },
+              "utter": {
+                "type": "string"
+              },
+              "rejections": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "required": [
+                    "if",
+                    "utter"
+                  ],
+                  "properties": {
+                    "if": {
+                      "type": "string"
+                    },
+                    "utter": {
+                      "type": "string"
+                    }
+                  }
+                }
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "link"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "link": {
+                "type": "string"
+              }
+            }
+          },
+          {
+            "required": [
+              "set_slots"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "reset_after_flow_ends": {
```

Comment:
I think we agreed for `set_slots` not to make this attribute accessible yet, I'd double-check with Engine/Daksh in case this changed?

Reply:
@dakshvar22 @twerkmeister Can we get a review of the step section at least 🙏

Reply:
Hm good point, never thought about this yet. I'd say let's drop it for now - it's not clear whether users want to always apply it to all slots in a set slot step and wether its useful at the end of a flow. Further, it just syntactic sugar, they could always have an explicit reset of the slots if they need it.

Reply:
To clarify as I'm getting confused - we had [this thread](https://rasa-hq.slack.com/archives/C047U7ZBKM1/p1695795629580879?thread_ts=1695632776.236379&cid=C047U7ZBKM1) about `reset_after_flow_ends` property for `set_slots` step, where the outcome was that this should be consistent with the default value for `collect` (True). I actually have a ticket I'm looking into today where I need to modify the current behaviour (because by default right now slots in a `set_slots` are not reset after flow ends). We can move this discussion to Slack, otherwise I think we're all in agreement that this property should not be part of the schema.

Reply:
I just had a look at the thread, and I think ultimately it was decided that for GA this is no supported. Let's gather some more hands-on and customer experience on this before implementing it.